debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisMemoryThreshold.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.redis;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.storage.redis.RedisClient;
+import io.debezium.util.IoUtil;
+import io.smallrye.mutiny.tuples.Tuple2;
+
+public class RedisMemoryThreshold implements Supplier<Boolean> {
```

Comment:
As `Supplier` is encapsulated in this class I don't think the class itself should be `Supplier` as it is always instantiated and used. There is no alternative computation.